# Exploring the Hotel-Level Data

Here, we convert all reservations to hotel nightly sales statistics using the `parse_dates` and `res_to_stats` functions in `utils.py`.

We'll start by deriving some basic information about each hotel, including:
* **Capacity** (total number of rooms)
* **Occupancy** (rooms sold / capacity)

Then, we'll pull more statistics into the `stats` DataFrames:
* **Revenue and Rooms Sold by Customer Segment**
* **ADR by Customer Segment**

These stats will help us understand what kind of hotels we're working with.

In [12]:
import pandas as pd
import numpy as np

from utils import generate_hotel_dfs

pd.options.display.max_rows = 30
pd.options.display.max_columns = 80

In [2]:
# h1_res, h1_dbd = generate_hotel_dfs('../data/H1.csv')

In [3]:
# h2_res, h2_dbd = generate_hotel_dfs('../data/H2.csv')

In [4]:
# Uncomment these four lines to save results from generate_hotel_dfs to save time later
# h1_dbd.to_pickle("pickle/h1_dbd.pick")
# h2_dbd.to_pickle("pickle/h2_dbd.pick")
# h1_res.to_pickle("pickle/h1_res.pick")
# h2_res.to_pickle("pickle/h2_res.pick")

# if you have already run generate_hotel_dfs and pickled the result, use this to save time.
h1_dbd = pd.read_pickle("pickle/h1_dbd.pick")
h2_dbd = pd.read_pickle("pickle/h2_dbd.pick")
h1_res = pd.read_pickle("pickle/h1_res.pick")
h2_res = pd.read_pickle("pickle/h2_res.pick")

In [14]:
h1_res.ArrivalDate.min()

Timestamp('2015-07-01 00:00:00')

In [17]:
len(h1_res)

40060

In [18]:
len(h2_res)

79330

In [15]:
h1_res.ArrivalDate.max()

Timestamp('2017-08-31 00:00:00')

In [16]:
h2_res

ResNum  IsCanceled  LeadTime  ArrivalDateYear ArrivalDateMonth  \
0           0           0         6             2015             July   
1           1           1        88             2015             July   
2           2           1        65             2015             July   
3           3           1        92             2015             July   
4           4           1       100             2015             July   
...       ...         ...       ...              ...              ...   
79325   79325           0        23             2017           August   
79326   79326           0       102             2017           August   
79327   79327           0        34             2017           August   
79328   79328           0       109             2017           August   
79329   79329           0       205             2017           August   

       ArrivalDateWeekNumber  ArrivalDateDayOfMonth  StaysInWeekendNights  \
0                         27                      1                     0   
1                         27                      1                     0   
2                         27                      1                     0   
3                         27                      1                     2   
4                         27                      2                     0   
...                      ...                    ...                   ...   
79325                     35                     30                     2   
79326                     35                     31                     2   
79327                     35                     31                     2   
79328                     35                     31                     2   
79329                     35                     29                     2   

       StaysInWeekNights  Adults  Children  Babies       Meal Country  \
0                      2       1       0.0       0  HB            PRT   
1                      4       2       0.0       0  BB            PRT   
2                      4       1       0.0       0  BB            PRT   
3                      4       2       0.0       0  BB            PRT   
4                      2       2       0.0       0  BB            PRT   
...                  ...     ...       ...     ...        ...     ...   
79325                  5       2       0.0       0  BB            BEL   
79326                  5       3       0.0       0  BB            FRA   
79327                  5       2       0.0       0  BB            DEU   
79328                  5       2       0.0       0  BB            GBR   
79329                  7       2       0.0       0  HB            DEU   

       MarketSegment DistributionChannel  IsRepeatedGuest  \
0      Offline TA/TO               TA/TO                0   
1          Online TA               TA/TO                0   
2          Online TA               TA/TO                0   
3          Online TA               TA/TO                0   
4          Online TA               TA/TO                0   
...              ...                 ...              ...   
79325  Offline TA/TO               TA/TO                0   
79326      Online TA               TA/TO                0   
79327      Online TA               TA/TO                0   
79328      Online TA               TA/TO                0   
79329      Online TA               TA/TO                0   

       PreviousCancellations  PreviousBookingsNotCanceled  ReservedRoomType  \
0                          0                            0  A                  
1                          0                            0  A                  
2                          0                            0  A                  
3                          0                            0  A                  
4                          0                            0  A                  
...                      ...                          ...               ...   
79325                      0              

In [7]:
h1_dbd.head()

DOW   Occ  RoomsSold    ADR   RoomRev  RevPAR  Trn_RoomsSold  \
2015-07-01  Wed  0.19       36.0  93.25   3356.83   17.95           30.0   
2015-07-02  Thu  0.34       64.0  97.68   6251.33   33.43           43.0   
2015-07-03  Fri  0.43       81.0  99.41   8052.35   43.06           54.0   
2015-07-04  Sat  0.58      108.0  96.98  10473.47   56.01           71.0   
2015-07-05  Sun  0.65      122.0  97.56  11901.72   63.65           79.0   

            Trn_ADR  Trn_RoomRev  Grp_RoomsSold  Grp_ADR  Grp_RoomRev  \
2015-07-01    95.91      2877.25            0.0      0.0          0.0   
2015-07-02   107.71      4631.34            0.0      0.0          0.0   
2015-07-03   109.70      5923.95            0.0      0.0          0.0   
2015-07-04   107.65      7643.08            0.0      0.0          0.0   
2015-07-05   105.76      8354.73            1.0    153.0        153.0   

            TrnP_RoomsSold  TrnP_ADR  TrnP_RoomRev  Cnt_RoomsSold  Cnt_ADR  \
2015-07-01             0.0      0.00          0.00            6.0    79.93   
2015-07-02             2.0     75.23        150.46           19.0    77.34   
2015-07-03             2.0     75.23        150.46           25.0    79.12   
2015-07-04             5.0     74.37        371.86           32.0    76.83   
2015-07-05             5.0     74.37        371.86           37.0    81.68   

            Cnt_RoomRev     WE     WD  STLY_Date  
2015-07-01       479.58  False   True 2014-07-02  
2015-07-02      1469.53  False   True 2014-07-03  
2015-07-03      1977.94   True  False 2014-07-04  
2015-07-04      2458.53   True  False 2014-07-05  
2015-07-05      3022.13  False   True 2014-07-06

In [8]:
[col for col in h1_dbd.columns]

['DOW',
 'Occ',
 'RoomsSold',
 'ADR',
 'RoomRev',
 'RevPAR',
 'Trn_RoomsSold',
 'Trn_ADR',
 'Trn_RoomRev',
 'Grp_RoomsSold',
 'Grp_ADR',
 'Grp_RoomRev',
 'TrnP_RoomsSold',
 'TrnP_ADR',
 'TrnP_RoomRev',
 'Cnt_RoomsSold',
 'Cnt_ADR',
 'Cnt_RoomRev',
 'WE',
 'WD',
 'STLY_Date']

In [9]:

h1_dbd

DOW   Occ  RoomsSold     ADR   RoomRev  RevPAR  Trn_RoomsSold  \
2015-07-01  Wed  0.19       36.0   93.25   3356.83   17.95           30.0   
2015-07-02  Thu  0.34       64.0   97.68   6251.33   33.43           43.0   
2015-07-03  Fri  0.43       81.0   99.41   8052.35   43.06           54.0   
2015-07-04  Sat  0.58      108.0   96.98  10473.47   56.01           71.0   
2015-07-05  Sun  0.65      122.0   97.56  11901.72   63.65           79.0   
...         ...   ...        ...     ...       ...     ...            ...   
2017-08-27  Sun  0.96      179.0  192.15  34394.38  183.93          125.0   
2017-08-28  Mon  0.93      174.0  190.89  33215.19  177.62          130.0   
2017-08-29  Tue  0.93      173.0  185.95  32169.28  172.03          130.0   
2017-08-30  Wed  0.93      174.0  176.95  30788.80  164.65          130.0   
2017-08-31  Thu  0.91      170.0  171.07  29082.20  155.52          130.0   

            Trn_ADR  Trn_RoomRev  Grp_RoomsSold  Grp_ADR  Grp_RoomRev  \
2015-07-01    95.91      2877.25            0.0     0.00         0.00   
2015-07-02   107.71      4631.34            0.0     0.00         0.00   
2015-07-03   109.70      5923.95            0.0     0.00         0.00   
2015-07-04   107.65      7643.08            0.0     0.00         0.00   
2015-07-05   105.76      8354.73            1.0   153.00       153.00   
...             ...          ...            ...      ...          ...   
2017-08-27   211.64     26454.38            5.0   152.61       763.03   
2017-08-28   205.76     26748.82            5.0   146.21       731.03   
2017-08-29   198.38     25789.27            5.0   146.21       731.03   
2017-08-30   187.56     24383.42            7.0   135.00       945.03   
2017-08-31   182.60     23738.62            6.0   148.12       888.73   

            TrnP_RoomsSold  TrnP_ADR  TrnP_RoomRev  Cnt_RoomsSold  Cnt_ADR  \
2015-07-01             0.0      0.00          0.00            6.0    79.93   
2015-07-02             2.0     75.23        150.46           19.0    77.34   
2015-07-03             2.0     75.23        150.46           25.0    79.12   
2015-07-04             5.0     74.37        371.86           32.0    76.83   
2015-07-05             5.0     74.37        371.86           37.0    81.68   
...                    ...       ...           ...            ...      ...   
2017-08-27            21.0    180.36       3787.50           28.0   121.05   
2017-08-28            13.0    205.13       2666.64           26.0   118.03   
2017-08-29            12.0    208.00       2496.04           26.0   121.27   
2017-08-30            12.0    210.97       2531.66           25.0   117.15   
2017-08-31            10.0    184.77       1847.66           24.0   108.63   

            Cnt_RoomRev     WE     WD  STLY_Date  
2015-07-01       479.58  False   True 2014-07-02  
2015-07-02      1469.53  False   True 2014-07-03  
2015-07-03      1977.94   True  False 2014-07-04  
2015-07-04      2458.53   True  False 2014-07-05  
2015-07-05      3022.13  False   True 2014-07-06  
...                 ...    ...    ...        ...  
2017-08-27      3389.47  False   True 2016-08-28  
2017-08-28      3068.70  False   True 2016-08-29  
2017-08-29      3152.94  False   True 2016-08-30  
2017-08-30      2928.69  False   True 2016-08-31  
2017-08-31      2607.19  False   True 2016-09-01  

[793 rows x 21 columns]

In [10]:
h1_dbd.describe()

Occ   RoomsSold         ADR       RoomRev      RevPAR  \
count  793.000000  793.000000  793.000000    793.000000  793.000000   
mean     0.804968  150.535939   89.606583  14532.738588   77.715460   
std      0.218965   40.949113   45.317853   9477.573530   50.682171   
min      0.160000   29.000000   37.880000   1215.570000    6.500000   
25%      0.700000  131.000000   54.580000   7403.050000   39.590000   
50%      0.920000  172.000000   70.780000  11888.980000   63.580000   
75%      0.960000  180.000000  111.900000  19720.260000  105.460000   
max      1.000000  187.000000  205.410000  37692.690000  201.570000   

       Trn_RoomsSold     Trn_ADR   Trn_RoomRev  Grp_RoomsSold     Grp_ADR  \
count     793.000000  793.000000    793.000000     793.000000  793.000000   
mean      102.303909   94.391463  10718.656772       0.916772   36.850643   
std        32.915559   48.123252   8122.392825       1.424225   53.303960   
min        11.000000   38.160000    858.700000       0.000000    0.000000   
25%        75.000000   55.620000   4542.660000       0.000000    0.000000   
50%       103.000000   77.440000   7614.720000       0.000000    0.000000   
75%       130.000000  121.440000  14543.530000       1.000000   56.000000   
max       163.000000  225.890000  31199.210000      11.000000  250.300000   

       Grp_RoomRev  TrnP_RoomsSold    TrnP_ADR  TrnP_RoomRev  Cnt_RoomsSold  \
count   793.000000      793.000000  793.000000    793.000000     793.000000   
mean     79.692005       30.133670   88.072333   2378.436999      17.181589   
std     159.492260       29.042111   51.216005   2024.231358      16.779235   
min       0.000000        0.000000    0.000000      0.000000       0.000000   
25%       0.000000       11.000000   54.000000    868.340000       2.000000   
50%       0.000000       19.000000   68.780000   1885.260000      12.000000   
75%      85.000000       41.000000  118.580000   3480.630000      30.000000   
max    1199.820000      162.000000  230.430000  12497.410000      74.000000   

          Cnt_ADR  Cnt_RoomRev  
count  793.000000   793.000000  
mean    63.564023  1355.952812  
std     35.211278  1335.427348  
min      0.000000     0.000000  
25%     44.070000    80.960000  
50%     61.390000   979.330000  
75%     80.010000  2531.630000  
max    146.340000  5293.090000

## Capacity

EW- ADD TO UTILS.PY

Based on the above tables, we can see the maximum capacity of each hotel.

**H1 (Resort Hotel)'s capacity is 187 rooms.**

**H2 (City Hotel)'s capacity is 226 rooms.**

## Get hotel statistics
* Occ, room rate, RevPAR, by month, by WD/WE
    * By segment?